Client/Frontend/Home/TopSitesPanel.swift

```diff
@@ -302,20 +308,25 @@ private class TopSitesLayout: UICollectionViewLayout {
 
     override init() {
         super.init()
-        setupForOrientation(UIApplication.sharedApplication().statusBarOrientation)
+        let rootWindow = (UIApplication.sharedApplication().delegate as! AppDelegate).window!
+        setupForTraitCollection(rootWindow.traitCollection, viewSize: rootWindow.bounds.size)
     }
 
     required init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
 
-    private func setupForOrientation(orientation: UIInterfaceOrientation) {
-        if orientation.isLandscape {
+    private func setupForTraitCollection(traitCollection: UITraitCollection, viewSize: CGSize) {
+        if traitCollection.userInterfaceIdiom == .Pad && traitCollection.verticalSizeClass == .Regular && traitCollection.horizontalSizeClass == .Regular {
+            if viewSize.width > viewSize.height {
+                thumbnailCols = 5
+            } else {
+                thumbnailCols = 4
+            }
+        } else if traitCollection.horizontalSizeClass == .Compact && traitCollection.verticalSizeClass == .Compact {
             thumbnailCols = 5
-        } else if UIScreen.mainScreen().traitCollection.horizontalSizeClass == .Compact {
-            thumbnailCols = 3
         } else {
-            thumbnailCols = 4
+            thumbnailCols = 3
```

Comment:
Going off the comment from above, could we dynamically calculate this value based on the width rather than hardcoding a bunch of values for every possible configuration? Something like width / preferredThumbnailSize, for example?